apis/field_error_test.go

```diff
@@ -338,55 +340,77 @@ can not use @, do not try`,
 			// }
 			// --> I expect path to be myList[i].foo
 
-			err := &FieldError{
+			return (&FieldError{
 				Message: "invalid field(s)",
 				Paths:   []string{"foo"},
-			}
+			}).ViaIndex(0).ViaField("bar").ViaIndex(2).ViaIndex(1).ViaField("baz").ViaIndex(3).ViaIndex(4).ViaField("boof")
+		}(),
+		want: "invalid field(s): boof[4][3].baz[1][2].bar[0].foo",
+	}, {
+		name: "manual index, using ViaFieldIndex",
+		err: func() *FieldError {
+			// Example, return an error in a loop:
+			// for i, item := spec.myList {
+			//   err := item.validate().ViaIndex(i).ViaField("myList")
+			//   if err != nil {
+			// 		return err
+			//   }
+			// }
+			// --> I expect path to be myList[i].foo
 
-			err = err.ViaIndex(0).ViaField("bar")
-			err = err.ViaIndex(2).ViaIndex(1).ViaField("baz")
-			err = err.ViaIndex(3).ViaIndex(4).ViaField("boof")
-			return err
+			return (&FieldError{
```

Comment:
I think this is less readable, I am simulating errors in the way they would be called in real code.

Reply:
I agree that the chaining is easier to read in 2-3 medium-length calls rather than chained more than 60 chars long.

(Purely a formatting issue.)

Reply:
Eh, matter of taste. Reverted.